src/engine/mod.rs

```diff
@@ -0,0 +1,659 @@
+                ));
+            }
+        }
+
+        //Validate Custom Prop defined in WarpgrapherConfig exists as a WarpgrapherResolver
+        let mut dyn_scalar_props: Vec<WarpgrapherProp> = Vec::new();
+        let mut props_with_validator: Vec<WarpgrapherProp> = Vec::new();
+
+        for m in config.model.iter() {
+            for p in m.props.iter() {
+                p.resolver
+                    .as_ref()
+                    .map_or((), |_| dyn_scalar_props.push(p.clone()));
+                p.validator
+                    .as_ref()
+                    .map_or((), |_| props_with_validator.push(p.clone()));
+            }
+        }
+        for dsp in dyn_scalar_props.iter() {
+            let resolver_name = dsp.resolver.as_ref().ok_or_else(|| {
+                Error::new(
+                    ErrorKind::ResolverNotFound(
+                        format!(
+                            "Failed to resolve custom prop: {prop_name}. Missing resolver name.",
+                            prop_name = dsp.name
+                        ),
+                        dsp.name.to_string(),
+                    ),
+                    None,
+                )
+            })?;
+            if !resolvers.contains_key(resolver_name) {
+                return Err(Error::new(
+                    ErrorKind::ResolverNotFound(
+                        format!(
+                            "Engine could not find a resolver for the custom prop: {prop_name}.",
+                            prop_name = dsp.name
+                        ),
+                        dsp.name.clone(),
+                    ),
+                    None,
+                ));
+            }
+        }
+
+        //Validate Custom Input Validator defined in WarpgrapherConfig exists as WarpgrapherValidator
+        for pwv in props_with_validator.iter() {
+            let validator_name = pwv.validator.as_ref().ok_or_else(|| {
+                Error::new(
+                    ErrorKind::ValidatorNotFound(
+                        format!(
+                            "Failed to find custom validator for prop: {prop_name}.",
+                            prop_name = pwv.name
+                        ),
+                        pwv.name.to_string(),
+                    ),
+                    None,
+                )
+            })?;
+            if !validators.contains_key(validator_name) {
+                return Err(Error::new(
+                    ErrorKind::ValidatorNotFound(
+                        format!(
+                            "Engine could not find a validator for the custom prop: {prop_name}.",
+                            prop_name = pwv.name
+                        ),
+                        pwv.name.clone(),
+                    ),
+                    None,
+                ));
+            }
+        }
+
+        // validation passed
+        Ok(())
+    }
+
+    pub fn execute(
+        &self,
+        req: Json<GraphQLRequest>, //TODO make generic
```

Comment:
Good idea, we should create a ticket to track this.

Reply:
Issue #33 created for this.